research-software-directory/RSD-as-a-service
# 556 orcid login

# Allow users to login with ORCID

Changes proposed in this pull request:

* Users can login with ORCID if they are on the whitelist in the database
* Add a page where admins can add or delete whitelisted ORCIDs
* Fix bug: if you add something to RSD_AUTH_PROVIDERS that we don't support, the sign in button doesn't show up

How to test:
* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale scrapers=0`
* Copy the new values from `.env.example` to `.env` and fill in the secret (contact us or create an integration on the [ORCID sandbox environment](https://sandbox.orcid.org/))
* Add `ORCID` to the `RSD_AUTH_PROVIDERS` env variable
* Also add something nonsensical to the `RSD_AUTH_PROVIDERS` env variable
* Login with (a sandbox) ORCID, this should fail as you're not whitelisted
* Login as an admin, add your ORCID to the whitelist (and play around on that page)
* Login with your whitelisted ORCID, this time it should work

Closes #556

PR Checklist:
